auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -33,20 +34,25 @@
  */
 public class IdpResponse implements Parcelable {
     private final User mUser;
+
     private final String mToken;
     private final String mSecret;
 
-    private final int mErrorCode;
+    @Nullable private final Exception mException;
```

Comment:
Maybe instead of having a raw `Exception` class here we could have `FirebaseUIException` that has a "cause" (the underlying `Exception`) and an `ErrorCode` field.

Then we wouldn't have to check `instanceof NetworkException` later and it gives us more room to expand.

Reply:
There we go, the `FirebaseUiException` thing is a much better way of doing it! In my PR, I have a bunch of exceptions extending Firebase SDK ones which felt really weird.

As for the error code thing, I was kinda hoping to move away from that. Ideally I would like us to make the exceptions part of our public API once they're stabilized so that we can provide specific, actionable errors. (We could check for developer errors and fail fast by returning an exception instead of logging stuff and hoping devs look at it.) Also, I don't want to deal with native Java serialization since we're passing this through activities 😊. What do you think?